### PR TITLE
Test suit adapted to changes in code base for autonested Tensordicts

### PR DESCRIPTION
**Pull request by ruleva1983**
## Description

Went through the tests to check which are failing for autonested cases and disabled those until a code base is adapted
to support the new use cases

## Motivation and Context

Related to issue #116 

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:
- [x] Tests 

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: test/test_tensordict.py [hunk 1]
```diff
@@ -541,11 +541,6 @@ def test_to_tensordict(self, td_name, device):
     def test_select(self, td_name, device, strict, inplace):
         torch.manual_seed(1)
         td = getattr(self, td_name)(device)
-        if td_name == "autonested_td":
-            pytest.skip(
-                "Select function is not yet designed for auto-nesting case."
-                " Skipping auto-nesting test case!!"
-            )
```
Comment (tcbegley) [inline test/test_tensordict.py:548]: I thought we hadn't adapted `select` yet, but this test is passing?

File: test/test_tensordict.py [hunk 2]
```diff
@@ -814,9 +823,15 @@ def test_apply_other(self, td_name, device, inplace):
...
         td = getattr(self, td_name)(device)
+        if td_name == "autonested_td":
+            pytest.skip(
+                "Test failing in auto-nested case. The assert_allclose_td  function not"
+                "designed for this case. Skipping!!"
+            )
         new_td = TensorDict({}, batch_size=td.batch_size, device=device)
-        for key, item in td.items():
-            new_td.set(key, item)
+        keys_view = _TensorDictKeysView(td, include_nested=True, leaves_only=False, error_on_loop=False)
```
Comment (tcbegley) [inline test/test_tensordict.py:832]: Why are we swapping `.items` for `_TensorDictKeysView` with `include_nested=True` here? This should only be necessary if we have `.items(include_nested=True)`.

File: test/test_tensordict.py [hunk 3]
```diff
@@ -1124,11 +1161,13 @@ def test_update(self, td_name, device, clone):
         assert set(td.keys()) == keys.union({"x"})
         # now with nested
         td["newnested"] = {"z": torch.zeros(td.shape)}
-        keys = set(td.keys(True))
+        keys_view = _TensorDictKeysView(td, include_nested=True, leaves_only=False, error_on_loop=False)
+        keys = list(keys_view)
+        keys = set(keys)
```
Comment (tcbegley) [inline test/test_tensordict.py:1166]:
```suggestion
        keys = set(keys_view)
```

File: test/test_tensordict.py [hunk 4]
```diff
@@ -1124,11 +1161,13 @@ def test_update(self, td_name, device, clone):
...
         assert ("newnested", "z") in keys
         td.update({"newnested": {"y": torch.zeros(td.shape)}}, clone=clone)
         keys = keys.union({("newnested", "y")})
-        assert keys == set(td.keys(True))
+        assert keys == set(list(keys_view))
```
Comment (tcbegley) [inline test/test_tensordict.py:1170]:
```suggestion
        assert keys == set(keys_view)
```

File: test/test_tensordict.py [hunk 5]
```diff
@@ -1137,14 +1176,10 @@ def test_update(self, td_name, device, clone):
             clone=clone,
         )
         keys = keys.union({("newnested", "x"), ("newnested", "w")})
-        assert keys == set(td.keys(True))
+        assert keys == set(list(keys_view))
```
Comment (tcbegley) [inline test/test_tensordict.py:1179]:
```suggestion
        assert keys == set(keys_view)
```

File: test/test_tensordict.py [hunk 6]
```diff
@@ -1137,14 +1176,10 @@ def test_update(self, td_name, device, clone):
...
         td.update({("newnested",): {"v": torch.zeros(td.shape)}}, clone=clone)
-        keys = keys.union(
-            {
-                ("newnested", "v"),
-            }
-        )
-        assert keys == set(td.keys(True))
+        keys = keys.union({("newnested", "v"),})
+        assert keys == set(list(keys_view))
```
Comment (tcbegley) [inline test/test_tensordict.py:1182]: (deleting the comma will stop the formatter exploding onto multiple lines)

```suggestion
        keys = keys.union({("newnested", "v")})
        assert keys == set(keys_view)
```

File: test/test_tensordict.py [hunk 7]
```diff
@@ -1678,7 +1728,7 @@ def test_nested_dict_init(self, td_name, device):
         td_clone["d"] = nested_tensordict_value
         # Re-init new TensorDict from dict, and check if they're equal
         td_dict_init = TensorDict(td_dict, batch_size=td.batch_size, device=device)
-
+        #
```
Comment (tcbegley) [inline test/test_tensordict.py:1731]:
```suggestion

```

**Review by tcbegley — CHANGES_REQUESTED** [review]
Thanks for working on this. Just a few things to tidy up.

Can you please also run pre-commit when committing. This will format your code for you and also alert you to any linting errors that need to be fixed.

```
pre-commit run --all-files
```

File: test/test_tensordict.py (hunk 1 above)
Comment (ruleva1983) [reply test/test_tensordict.py:548]: Now seems yes

File: test/test_tensordict.py (hunk 3 above)
Comment (ruleva1983) [reply test/test_tensordict.py:1166]: Ok

File: test/test_tensordict.py (hunk 4 above)
Comment (ruleva1983) [reply test/test_tensordict.py:1170]: Ok

File: test/test_tensordict.py (hunk 2 above)
Comment (ruleva1983) [reply test/test_tensordict.py:832]: Reverted

**Review by tcbegley — APPROVED** [review]
LGTM

**Pull request merged**